syntax = "proto3";
package serialization;

import "map_renderer.proto";
import "transport_router.proto";

message Coordinates {
	double lat = 1;
	double lng = 2;
}

message Stop {
	string name = 1;
	Coordinates coordinates = 2;
}

message StopDistance {
	uint32 stop1 = 1;
	uint32 stop2 = 2;
	uint32 distance = 3;
}
message Bus {
	string name = 1;
	repeated uint32 route = 2;
	bool isRoundtrip = 3;
}

message TransportCatalogue {
	repeated Stop stops = 1;
	repeated Bus buses = 2;
	repeated StopDistance stopDistances = 3;
}

message Base {
	TransportCatalogue catalogue = 1;
	Map_renderer renderer = 2;
	Transport_router router = 3;
}
